src/docs/guides/rails.md

```diff
@@ -0,0 +1,200 @@
+
+5. **Set Up a Public URL**:
+    - Navigate to the **Networking** section under the [Settings](/overview/the-basics#service-settings) tab of your new service.
+    - Click [Generate Domain](/guides/public-networking#railway-provided-domain) to create a public URL for your app.
+
+<Image src="https://res.cloudinary.com/railway/image/upload/f_auto,q_auto/v1728049734/docs/quick-start/railsapp_on_railway.png"
+alt="screenshot of the deployed Rails service showing the Hello world page"
+layout="responsive"
+width={2375} height={1151} quality={100} />
+
+This guide covers the main deployment options on Railway. Choose the approach that suits your setup, and start deploying your Rails apps effortlessly!
+
+## Set Up Workers & Cron Jobs with Sidekiq
+
+Sidekiq is a powerful and efficient background job processor for Ruby apps, and it integrates seamlessly with Rails. Follow the instructions below to configure and run Sidekiq in your Rails app on Railway:
+
+1. **Install Sidekiq**
+    - Start by adding `sidekiq` and `sidekiq-cron` to your Rails app. In your terminal, run the following command:
+        ```bash
+        bundle add sidekiq
+        bundle add sidekiq-cron
+        ```
+2. **Add a Redis Database Service**
+    - Sidekiq uses Redis as a job queue. To set this up:
+        - Right-click on the Railway dashboard canvas or click the **Create** button.
+        - Select **Database**.
+        - Select **Add Redis** from the available databases.
+            - This will create and deploy a new Redis service for your app.
+3. **Create and Configure a Worker Service**
+    - Now, set up a separate service to run your Sidekiq workers.
+        - Create a new **Empty Service** and name it **Worker Service**.
+        - Go to the <a href="/overview/the-basics#service-settings">**Settings**</a> tab of this service to configure it.
+        - In the **Source** section, connect your GitHub repository to the **Source Repo**.
+        - Under the <a href="/guides/build-configuration#customize-the-build-command">**Build**</a> section, set `bundle install` as the **Custom Build Command**. This installs the necessary dependencies.
+        - In the **Deploy** section, set `bundle exec sidekiq` as the <a href="/guides/start-command">**Custom Start Command**</a>. This command will start Sidekiq and begin processing jobs.
+        - Click on <a href="/overview/the-basics#service-variables">**Variables**</a> at the top of the service settings.
+        - Add the following environment variables:
+            - `SECRET_KEY_BASE` or `RAILS_MASTER_KEY`: Set this to the value of your Rails app’s secret key.
+            - `REDIS_URL`: Set this to `${{Redis.REDIS_PUBLIC_URL}}` to reference the Redis database URL. This tells Sidekiq where to find the job queue. Learn more about [referencing service variables](/guides/variables#referencing-another-services-variable).
```

Comment:
Public URL?

Reply:
Remember the convo we had about talking to the database using private networking? It doesn't work during database migration on every deploy so I simply use the public URL which works seamlessly.

Reply:
Can Rails be made to do migrations during runtime before starting the app?

Reply:
I'll try this out and revert

Reply:
This works perfectly on runtime! Updated the guide to have the relevant knowledge. @brody192